### PR TITLE
CLOUDP-178753: Deletion protection for AtlasBackupSchedule and AtlasBackupPolicy

### PR DESCRIPTION
**Pull request by igor-karpukhin**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: pkg/api/v1/atlasbackupschedule_types_test.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+			},
+			shouldFail: false,
+		},
+	}
+
+	for _, tt := range testData {
+		result := tt.inSchedule.ToAtlas(tt.output.ClusterID, tt.clusterName, tt.inPolicy)
+		if !reflect.DeepEqual(result, tt.output) {
+			exp, _ := json.MarshalIndent(tt.output, "", " ")
+			got, _ := json.MarshalIndent(result, "", " ")
```
Comment (josvazg) [inline pkg/api/v1/atlasbackupschedule_types_test.go:80]: why not comparing the go structs directly?

Comment (igor-karpukhin) [reply pkg/api/v1/atlasbackupschedule_types_test.go]: Found a better alternative: https://github.com/go-test/deep

File: pkg/api/v1/atlasbackupschedule_types.go [hunk 2]
```diff
@@ -96,6 +99,51 @@ type AtlasBackupSchedule struct {
...
+		result.CopySettings = append(result.CopySettings, mongodbatlas.CopySetting{
+			CloudProvider:     copySetting.CloudProvider,
+			RegionName:        copySetting.RegionName,
+			ReplicationSpecID: copySetting.ReplicationSpecID,
+			ShouldCopyOplogs:  copySetting.ShouldCopyOplogs,
+			Frequencies:       copySetting.Frequencies,
+		})
+	}
+
+	result.ClusterID = clusterID
```
Comment (helderjs) [inline pkg/api/v1/atlasbackupschedule_types.go:143]: Out of curiosity, Why are you setting the cluster ID here and not in the block 114-124? Also, both ID and name are needed?

File: pkg/api/v1/atlasbackupschedule_types_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	"testing"
+
+	"github.com/go-test/deep"
+	"go.mongodb.org/atlas/mongodbatlas"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/toptr"
+)
+
+func Test_BackupScheduleToAtlas(t *testing.T) {
+	testData := []struct {
```
Comment (helderjs) [inline pkg/api/v1/atlasbackupschedule_types_test.go:13]: I always suggest individual subtests in favor of table tests in cases with dense data and reduced readability.

File: pkg/controller/atlasdeployment/backup.go [hunk 4]
```diff
@@ -26,6 +27,10 @@ import (
 	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
 )
 
+var errArgIsNotBackupSchedule = errors.New("failed to match resource type as AtlasBackupSchedule")
+
+const BackupProtected = "unable to reconcile AtlasBackupSchedule: it already exists in Atlas, it was not previously managed by the operator, and the deletion protection is enabled"
```
Comment (helderjs) [inline pkg/controller/atlasdeployment/backup.go:32]: The error message is standardized among the resources and must point to the docs
```unable to reconcile RESOURCE_NAME due to deletion protection being enabled. see https://dochub.mongodb.org/core/ako-deletion-protection for further information```

**Review by helderjs — CHANGES_REQUESTED** [review]

File: pkg/api/v1/atlasbackupschedule_types.go (hunk 2 above)
Comment (igor-karpukhin) [reply pkg/api/v1/atlasbackupschedule_types.go]: Good catch! This is a leftover from the previous implementation. Fixed

File: pkg/api/v1/atlasbackupschedule_types_test.go (hunk 3 above)
Comment (igor-karpukhin) [reply pkg/api/v1/atlasbackupschedule_types_test.go]: A matter of taste I think, but converted to a sub-test

File: pkg/controller/atlasdeployment/backup.go (hunk 4 above)
Comment (igor-karpukhin) [reply pkg/controller/atlasdeployment/backup.go]: Thanks! Fixed

**Review by helderjs — APPROVED** [review]
Good job!

**Pull request merged**